src/main/kotlin/com/gamenews/plugins/Controller.kt

```diff
@@ -14,24 +15,39 @@ import io.ktor.util.pipeline.PipelineContext
 /**
  * This class handles the calls for all routes
  */
-class Controller {
+class Controller(
+    private val db: ArticlesDatabase
+) {
+
     /**
      * Handles calls to show all articles
      */
-    suspend fun showAllArticles(context: PipelineContext<Unit, ApplicationCall>) {
+    suspend fun getAllArticles(context: PipelineContext<Unit, ApplicationCall>) {
+        val allArticles = db.getAllArticles()
+
+        if (allArticles.isEmpty()) {
+            context.call.respond(
+                HttpStatusCode.NotFound,
+                "Something went terribly wrong. You have no articles!"
+            )
+            return
+        }
+
         context.call.respond(
+            HttpStatusCode.OK,
             FreeMarkerContent(
                 "index.ftl",
-                mapOf("articles" to articles)
+                mapOf("articles" to allArticles)
             )
         )
     }
 
     /**
      * Handles calls for creating a new article
      */
-    suspend fun newArticle(context: PipelineContext<Unit, ApplicationCall>) {
+    suspend fun createNewArticle(context: PipelineContext<Unit, ApplicationCall>) {
         context.call.respond(
```

Comment:
We need to call the database here to create the new article.

Reply:
Or I think that `saveArticle` is used for creating a new article? (thinking face). This displays the page to create a new article right?

Reply:
Done